ui/app/pages/swaps/build-quote/build-quote.js

```diff
@@ -132,7 +133,7 @@ export default function BuildQuote ({
         .then((fetchedBalance) => {
           if (fetchedBalance?.balance) {
             const balanceAsDecString = fetchedBalance.balance.toString(10)
-            const balanceAsHexString = fetchedBalance.balance.toString(16)
+            const balanceAsHexString = addHexPrefix(fetchedBalance.balance.toString(16))
             const userTokenBalance = calcTokenAmount(balanceAsDecString, token.decimals)
             dispatch(setSwapsFromToken({ ...token, string: userTokenBalance.toString(10), balance: balanceAsHexString }))
```

Comment:
Why is `balance` being set to a hex string here? The `balance` property of a token should be a decimal string.

Reply:
right :facepalm:

Reply:
I noticed that `ethBalance` is being passed in here somewhere as well, which is the the Ether balance in WEI as a hex string.That's a similar case that is also wrong.

Reply:
the `TokenOverview` page is calling `setSwapsFromToken` with no `balance` as well :thinking:

Reply:
These second two issues are addressed in https://github.com/MetaMask/metamask-extension/pull/9588